src/runtime_src/core/common/drv/include/kds_core.h

```diff
@@ -210,6 +211,22 @@ int kds_add_context(struct kds_sched *kds, struct kds_client *client,
 		    struct kds_client_cu_ctx *cu_ctx);
 int kds_del_context(struct kds_sched *kds, struct kds_client *client,
 		    struct kds_client_cu_ctx *cu_ctx);
+/* End of legacy context functions */
+
+/* Start of hw context functions */
```

Comment:
I will suggest you have a new set of APIs in this section.
If some APIs are implemented in the legacy functions, move the implementation to new functions. In legacy function, call the new one.

In clean up stage, delete old data structure and old APIs in the header file, then go fix all of the compile issue.

Reply:
Will take care in Future PRs. Thanks for the suggestion.